### DISCUSSION TITLE
counting consecutive numbers from a list in python

### DISCUSSION THREAD

**Discussion by TekleAman** [jertel/elastalert2#781] [Q&A] [answered]
Hi everyone,
I am new to github and I am not sure if I can ask code question here. Please let me know if this is not the right place.

To my question: 
I have a list of numbers that composed of zeros, negative and positive numbers. The goal is to count consecutive negative numbers including preceding  and succeeding zeros, a similar will be performed to positive numbers. I tried using the itertools but so far I did not get the desire result. 
The dedired output from the list ( val) below will be  [7,2,7,4,3]  
    val = [0,0,0,0,-1,-2,-3,4,5,-6,0,0,0,-7,-8,-9,10,0,11,10,-1,-2,0]

   import itertools

   for _, v in itertools.groupby(val, lambda x :   x<0 or x ==0):

      a = [m for  m  in v ]
      print(a)

**Comment by nsano-rururu** [comment, marked as answer]
It's not a site that answers how to write Python code. We recommend searching on Google or buying a Python book.
